gmail/src/views/company.ts

```diff
@@ -5,32 +5,53 @@ import { createKeyValueWidget, actionCall, notify } from "./helpers";
 import { URLS } from "../const";
 import { State } from "../models/state";
 import { Company } from "../models/company";
+import { Partner } from "../models/partner";
 import { _t } from "../services/translation";
 
+function onCreateCompany(state: State) {
+    const [company, error] = Partner.createCompany(state.partner.id);
+    if (company) {
+        state.partner.company = company;
+    }
+    state.error = error;
+    return updateCard(buildView(state));
+}
+
+function onEnrichCompany(state: State) {
+    const [company, error] = Partner.enrichCompany(state.partner.company.id);
+    if (company) {
+        state.partner.company = company;
+    }
+    state.error = error;
+    return updateCard(buildView(state));
+}
+
 export function buildCompanyView(state: State, card: Card) {
     if (state.partner.company) {
         const odooServerUrl = State.odooServerUrl;
         const company = state.partner.company;
 
         const companySection = CardService.newCardSection().setHeader("<b>" + _t("Company") + "</b>");
 
-        const companyContent = [company.email, company.phone]
-            .filter((x) => x)
-            .map((x) => `<font color="#777777">${x}</font>`);
-
-        companySection.addWidget(
-            createKeyValueWidget(
-                null,
-                company.name + "<br>" + companyContent.join("<br>"),
-                company.image || UI_ICONS.no_company,
-                null,
-                null,
-                company.id ? odooServerUrl + `/web#id=${company.id}&model=res.partner&view_type=form` : null,
-                false,
-                company.email,
-                CardService.ImageCropType.CIRCLE,
-            ),
-        );
+        if (!state.partner.id || state.partner.id !== company.id) {
```

Comment:
What is the purpose of this if actually ?

Reply:
The user might open an email of the company itself, in that case the information are redundant because you can see the company information on the "contact" section and also on the "company" section, so with this we hide the redundant information on the company section